subsys/shell/Kconfig.backends

```diff
@@ -117,6 +117,7 @@ config SHELL_BACKEND_RTT
 	select USE_SEGGER_RTT
```

Comment:
if `depends on HAS_SEGGER_RTT` is removed then this should be changed to `depends on USE_SEGGER_RTT`. This option should appear only if rtt is used.

Reply:
Changed.

Wasn't sure if there might be configuration files setting only `SHELL_BACKEND_RTT` without setting `USE_SEGGER_RTT`. None in-tree at least.

Please review at https://github.com/zephyrproject-rtos/zephyr/pull/22075 btw. This PR is just a dumping ground for cherry-picked fixes while testing.